### PR TITLE
Residence refactor

### PR DESCRIPTION
**Pull request by ratneshraval**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- All existing tests updated to pass

#### Which User story does this relate to?
- https://www.pivotaltracker.com/story/show/148091485 (WIP)
- https://www.pivotaltracker.com/story/show/148679103
- https://www.pivotaltracker.com/story/show/148766707

#### Brief feature description
- Address card refactor to bring state up to index. Some other bug fixes

### CODE REVIEW

File: app/javascript/forms/addressCard.js [hunk 1]
```diff
@@ -1,112 +1,104 @@
...
+  city: '',
+  state: {
+    id: '',
+    value: ''
+  },
+  type: {
+    id: '2',
+    value: 'Mailing'
+  }
+})
```
Comment (lauherkcambria) [inline app/javascript/forms/addressCard.js]: there has got to be a better way to do this -- why not pass in the  type value via a method? i.e. ```
defaultFields(type) {
return {
...,
type:
{
id:'2',
 value: type
}
}
} 
```

Comment (lauherkcambria) [inline app/javascript/forms/addressCard.js]: also, this breaks the rule of one component doing 1 thing

File: app/javascript/constants/constants.js [hunk 2]
```diff
@@ -63,11 +63,11 @@ const CountyList = Object.freeze({
 const yesNo = Object.freeze({
   'items': [
     {
-      'id': 1,
+      'id': true,
       'value': 'Yes'
     },
     {
-      'id': 2,
+      'id': false,
```
Comment (lauherkcambria) [inline app/javascript/constants/constants.js]: why? usually with hashes id's aren't booleans

File: app/javascript/forms/addressCard.js (hunk 1 above)
Comment (ratneshraval) [inline app/javascript/forms/addressCard.js]: I agree. I don't like how the address isn't in its own component, which is resulting in complicated address update code. 
I added a new story to refactor and move address to it's own component. https://www.pivotaltracker.com/story/show/149177125

File: app/javascript/constants/constants.js (hunk 2 above)
Comment (ratneshraval) [inline app/javascript/constants/constants.js]: Yes/no dropdowns require us to send boolean value to API, so in this case the ID is boolean and text is yes/no.

File: app/javascript/forms/addressCard.js (hunk 1 above)
Comment (lauherkcambria) [inline app/javascript/forms/addressCard.js]: nice 👍

File: app/javascript/constants/constants.js (hunk 2 above)
Comment (lauherkcambria) [inline app/javascript/constants/constants.js]: ah, gotcha. good to know! 👍

**Pull request merged**